Marlin/Marlin_main.cpp

```diff
@@ -4295,7 +4295,7 @@ inline void gcode_M109() {
     #define TEMP_CONDITIONS (!residency_start_ms || PENDING(now, residency_start_ms + (TEMP_RESIDENCY_TIME) * 1000UL))
   #else
     // Loop until the temperature is very close target
-    #define TEMP_CONDITIONS (isHeatingHotend(target_extruder))
+    #define TEMP_CONDITIONS (no_wait_for_cooling ? isHeatingHotend(target_extruder) : isCoolingHotend(target_extruder))
```

Comment:
@jbrazio L4286 is the condition "_Exit if the temperature is above target and not waiting for cooling_". If M109/M190 is entered whilst the temperature is above the set target using the 'S' parameter then it will escape right away.

For example if you did the following:
````
M109 S120 ; this waits until 120°C is reached
M109 S100 ; this will set the target to 100°C and escape right away
````